test/conformance/loadbalancing/feature.go

```diff
@@ -0,0 +1,138 @@
+func InitializeScenario(ctx *godog.ScenarioContext) {
+	ctx.Step(`^a new random namespace$`, aNewRandomNamespace)
+	ctx.Step(`^an Ingress resource named "([^"]*)" with this spec:$`, anIngressResourceNamedWithThisSpec)
+	ctx.Step(`^The Ingress status shows the IP address or FQDN where it is exposed$`, theIngressStatusShowsTheIPAddressOrFQDNWhereItIsExposed)
+	ctx.Step(`^The backend deployment "([^"]*)" for the ingress resource is scaled to (\d+)$`, theBackendDeploymentForTheIngressResourceIsScaledTo)
+	ctx.Step(`^I send (\d+) requests to "([^"]*)"$`, iSendRequestsTo)
+	ctx.Step(`^all the responses status-code must be (\d+) and the response body should contain the IP address of (\d+) different Kubernetes pods$`, allTheResponsesStatuscodeMustBeAndTheResponseBodyShouldContainTheIPAddressOfDifferentKubernetesPods)
+
+	ctx.BeforeScenario(func(*godog.Scenario) {
+		state = tstate.New()
+		resultStatus = make(map[int]sets.String, 0)
+	})
+
+	ctx.AfterScenario(func(*messages.Pickle, error) {
+		// delete namespace an all the content
+		_ = kubernetes.DeleteNamespace(kubernetes.KubeClient, state.Namespace)
+	})
+}
+
+func aNewRandomNamespace() error {
+	ns, err := kubernetes.NewNamespace(kubernetes.KubeClient)
+	if err != nil {
+		return err
+	}
+
+	state.Namespace = ns
+	return nil
+}
+
+func anIngressResourceNamedWithThisSpec(name string, spec *messages.PickleStepArgument_PickleDocString) error {
+	ingress, err := kubernetes.IngressFromSpec(name, state.Namespace, spec.GetContent())
+	if err != nil {
+		return err
+	}
+
+	err = kubernetes.DeploymentsFromIngress(kubernetes.KubeClient, ingress)
+	if err != nil {
+		return err
+	}
+
+	err = kubernetes.NewIngress(kubernetes.KubeClient, state.Namespace, ingress)
+	if err != nil {
+		return err
+	}
+
+	state.IngressName = name
+
+	return nil
+}
+
+func theIngressStatusShowsTheIPAddressOrFQDNWhereItIsExposed() error {
+	ingress, err := kubernetes.WaitForIngressAddress(kubernetes.KubeClient, state.Namespace, state.IngressName)
+	if err != nil {
+		return err
+	}
+
+	state.IPOrFQDN = ingress
+	return err
+}
+
+func iSendRequestsTo(totalRequest int, rawURL string) error {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return err
+	}
+
+	for iteration := 1; iteration <= totalRequest; iteration++ {
+		capturedRequest, capturedResponse, err := http.CaptureRoundTrip("GET", u.Scheme, u.Host, u.Path, state.IPOrFQDN)
+		if err != nil {
+			return err
+		}
+
+		if resultStatus[capturedResponse.StatusCode] == nil {
+			resultStatus[capturedResponse.StatusCode] = sets.NewString()
+		}
+
+		resultStatus[capturedResponse.StatusCode].Insert(capturedRequest.Pod)
+	}
+
+	return nil
+}
+
+func allTheResponsesStatuscodeMustBeAndTheResponseBodyShouldContainTheIPAddressOfDifferentKubernetesPods(statusCode int, pods int) error {
+	results, ok := resultStatus[statusCode]
+	if !ok {
+		return fmt.Errorf("expected %v different POD ip addresses for status code %v but none was returned", statusCode, pods)
```

Comment:
This error string looks wrong given the format and passed variables.

The condition also seems to ignore the fact that some (of the 100) responses might be something other than 200, as long as each pod responded at least once with 200.

Reply:
changed